src/cargo/core/manifest.rs

```diff
@@ -662,6 +667,7 @@ impl Target {
 
     pub fn bin_target(
```

Comment:
I think this is something that we'll want to handle for at least the cdylib output as well, perhaps the staticlib output too. Overall I think that the `filename` handling from the TOML profile may need to happen at a lower layer than as a new method on constructors here (as otherwise it would need to be added as an argument to all constructors)

Reply:
I would prefer to re-use the `name` field in the same struct, instead of creating a new field as I have done here. But, @ehuss, suggested not to do that as we don't know what kind of implications that might have.